components/util/Anchors.tsx

```diff
@@ -73,9 +87,62 @@ export function AnchorsSetter(props) {
 
   useEffect(() => {
     setAnchors(anchors);
+    setActiveAnchor(anchors[0]);
 
-    return () => setAnchors([]);
+    return () => {
+      setAnchors([]);
+      setActiveAnchor(null);
+    };
   }, [router?.pathname]);
 
+  useLayoutEffect(() => {
+    let raf: number;
+    const anchorEls = anchors.map((a: Anchor) => document.getElementById(a.id));
+
+    const handleScroll = () => {
+      raf = requestAnimationFrame(() => {
```

Comment:
Here's where we set the `raf` variable. Every time `handleScroll` is called, we pass a function to `requestAnimationFrame` that we want to run every animation frame (instead of running it immediately since `handleScroll` gets called a lot).